arelle/Locale.py

```diff
@@ -74,14 +76,27 @@ def getUserLocale(localeCode: str = '') -> tuple[LocaleDict, str | None]:
                 conv = locale.localeconv()
             except locale.Error:
                 pass
-    locale.setlocale(locale.LC_ALL, 'C')
     if conv is None:  # some other issue prevents getting culture code, use 'C' defaults (no thousands sep, no currency, etc)
+        locale.setlocale(locale.LC_ALL, 'C')
         localeSetupMessage = f"locale code \"{localeCode}\" is not available on this system"
         conv = locale.localeconv() # use 'C' environment, e.g., en_US
-    if C_LOCALE is None:  # load culture-invariant C locale
-        C_LOCALE = locale.localeconv()
     return cast(LocaleDict, conv), localeSetupMessage
 
+
+def getUserLocale(localeCode: str = '') -> tuple[LocaleDict, str | None]:
```

Comment:
In initial discussion, we talked about how this change was most critical when Arelle is being used as a library, and not as critical when being used independently via GUI or command line. This implied that we might skip resetting the locale in those contexts.

Based on what I'm seeing, calls to `getUserLocale` are used sparingly in those contexts, generally once per run during initialization or on demand from the GUI. Because of this, I don't think it's worth the complexity to conditionally perform the reset based on the running context, and instead just commit to it being reset every time Arelle interacts with it.

Reply:
My concern with no longer setting the locale for the application context is that it could cause a regression somewhere that depends on the locale being `C` (default file encoding, time str formating, etc.) which Arelle currently sets it to during initialization. Admittedly I couldn't find anywhere in Arelle where this would be an issue, but I still think it's reasonable to set the locale to `C` when it's deployed as an app (and avoid potentially introducing a regression for one of our many builds/platforms.)

There are a couple ways to handle that.
1. Only set for the frozen builds:
```python
if getattr(sys, "frozen", False):
    locale.setlocale(locale.LC_ALL, 'C')
```
2. Set the locale to `C` from the application entry points ([`CntlrCmdLine`](https://github.com/Arelle/Arelle/blob/1a294dcf6d6aa05f9fb2de698a8fd72f5003f0fc/arelle/CntlrCmdLine.py#L548) and [`CntlrWinMain`](https://github.com/Arelle/Arelle/blob/1a294dcf6d6aa05f9fb2de698a8fd72f5003f0fc/arelle/CntlrWinMain.py#L63))

I'd lean towards option #2 as it covers running arelle as an app from source.

Reply:
My approach did cause regressions, because the conformance suites are now running without locale being set to `C`. This initially caused some failures due to some seemingly unnecessary localization of date/time parts. My last commit corrects these, but having `C` set in the application contexts should protect us from any unknown regressions that the conformance suites don't cover.